### DISCUSSION TITLE
Can't add custom resolver to enum

### DISCUSSION THREAD

**Discussion by sapinheiro** [graphql-java-kickstart/graphql-spring-boot#852] [Q&A]
It seems like the scenario I'm outlining isn't possible

I currently have (and this works!)
```
//graphql
type MyEnum {
   id
   value1
   value2 
   value3
   customFields
}

//java code

public Enum MyEnum {
     ONE(1, "one", "uno", "I"),
     TWO(2, "two", "dos", "II);
}
public MyEnumResolver implements GraphQLResolver<MyEnum> {

     public List<CustomFields> customFields(MyEnum enum) {
        return fieldGenerator.call(enum);
     }
}
```

But I also want to be able to use my enums as an input for other mutations and inputs (can't get this to work)

```
enum MyEnum {
   ONE
   TWO 
   THREE
}

input DeleteWorld {
   id: ID!
   val: MyEnum!
}

public MyEnumResolver implements GraphQLResolver<MyEnum> {
     public List<CustomFields> customFields(MyEnum enum) {
        return fieldGenerator.call(enum);
     }
}
```

It doesn't seem to be possible to use a pure graphql java enum and have extra values/custom resolvers. Is that correct?

**Comment by oliemansm** [comment]
Resolvers are only used to resolve values for responses, never for requests. If you have some custom field logic you need to apply to resolve certain values on input objects then you'll have to execute that manually.

**Comment by sapinheiro** [comment]
That's not quite what I mean. I want to be able to use the enum as an input type, but I also want to be able to resolve values for responses.

**Reply by oliemansm** [reply]
I notice just now that you have a `type MyEnum` and a `enum MyEnum`. Can you try renaming one of those, because I can imagine that resulting in a conflict.

**Reply by sapinheiro** [reply]
The first example is what I have, the second example is what I want. I have it as a type because I want to be able to resolve `customFields` (and the other values) but I also want to be able to use it as a traditional enum in graphql

**Reply by oliemansm** [reply]
Could be there's still a bug that it can't find it because it's part of an input object. Try adding a query to the schema that simply returns the enum. That should trigger the parser in adding the enum and being able to apply it for the input too.

**Reply by sapinheiro** [reply]
I'm not sure if I'm being clear or not. Fundamentally my question is: Given a pure graphql enum

```
enum MyEnum {
  ONE,
  TWO,
  THREE
}
```

is it possible to resolve custom values on top of it? If so, how do I do it in java and what would the graphql schema look like?

**Reply by sapinheiro** [reply]
Alternatively, is it possible to have both a type and enum that are tied to the same java class?

```
type MyEnum {
   id
   value1
   value2 
   value3
   customFields
}

// fails because it can't be the same name as the type. If I rename, it also fails because it can't a java class that matches to it
enum MyEnum {
  ONE,
  TWO,
  THREE
}

// java
public Enum MyEnum {
     ONE(1, "one", "uno", "I"),
     TWO(2, "two", "dos", "II);
}
```
